### ISSUE TITLE
Some confusion regarding text-image aligment and the loss function

### ISSUE THREAD

**Issue by HilbertXu** [Youncy-Hu/MAGE#2]
Hi there!

Thanks for your amazing work!
I'm new to this TI2V area and found your paper, it is amazingly insightful!
and also I have to thank you for open-sourcing your codes! They are really clear and well-structured!

However, I still have some confusions.

1. Regarding the text-image alignment.
In section 3.3.1 
" **We employ a learnable text encoder to compute the text embedding es ∈ R L×d from the input text s, where d is the hidden size. es is used as the key and the value of the cross-attention module. The image embedding z1 is converted to the same latent space as es by a learnable embedding matrix, and the converted embedding, denoted by ez1 ∈ R h×w×d , is used as the query.**"

Here you mentioned the image embeddings are converted to the space of text embedding. However, from your 
code, you use two different nn.Embedding for image and text respectively, they are "visual_token_embedding" and "token_embedding". Could you please explain a little bit in depth how you align these two embeddings together? I may misunderstand some parts of the code, if so please forgive me for my mistake.

2. Regarding the loss function
Could you please briefly explain why you choose the CE on the probabilistic distribution across the codebook as loss function rather than using the common L2 loss on reconstructed images and gt images? 
I think this is quite interesting because I was also thinking about using the same loss function.

Thanks in advance for your reply!

Best,
Yucheng

**Comment by Youncy-Hu** [comment]
Hi Yucheng,

Thanks for your asking. There are some answers I can give you.
1. The original inputs z_1 and text are token indices of visual codebook and text vocabulary, respectively. Before input to subsequent neural networks, it needs to be transferred into embedding vectors via an Embedding layer. Besides, since they are from two modalities, we should use two different embedding layers for them. The text-image alignment is achieved with cross-attention operation along with training process.
2. The training objective of MAGE is to predict codebook index for each token in latent space, which is a classification probolem, instead of generating pixel values in RGB space. Therefore, we predict the probabilistic distribution over codebook for each token and choose CE loss that suits for classification.

I hope that answers your questions. Have a nice day! :)
